lib/src/nuvigator.dart

```diff
@@ -73,7 +73,7 @@ class Nuvigator<T extends Router> extends Navigator {
               ?.fallbackScreenType(screenType)
               ?.toRoute(settings),
           key: key,
-          initialRoute: initialRoute,
+          initialRoute: router.getInitialRoute(initialRoute),
```

Comment:
I would create a second option: `initialDeepLink` and the user would be able to provide either an `initialRoute` OR an `initialDeepLink` (we can make an assert for that).
I think that this makes the API clearer of what to expect to pass in each case.